### PR TITLE
Unvendor sqlite3 from stdlib

### PR DESCRIPTION
**Pull request by ryanking13**
### Description

Unvendor the standard library sqlite3 to reduce the size of the main module. It reduces the size of `pyodide.asm.wasm` around ~1.4MB.

Related: #2940, #1542

__Before:__

```sh
du ../dist-main/pyodide*
4944    pyodide.asm.data
1516    pyodide.asm.js
8616    pyodide.asm.wasm
```

__After:__

```sh
$ du pyodide*
4944    pyodide.asm.data
1468    pyodide.asm.js
7264    pyodide.asm.wasm
$ du sqlite3*
468    sqlite3-1.0.0.zip
```

### Checklists

- [x] Add a [CHANGELOG](https://github.com/pyodide/pyodide/blob/main/docs/project/changelog.md) entry
- [x] Add / update tests
- [x] Add new / update outdated documentation

### CODE REVIEW

**Comment by tiran** [comment]
Emscripten 3.1.15 and newer come with sqlite3 in emports, `-sUSE_SQLITE3`.

**Comment by ryanking13** [comment]
> Emscripten 3.1.15 and newer come with sqlite3 in emports, `-sUSE_SQLITE3`.

Yes, thanks for your porting work @tiran! For now, we are stuck in Emscripten 3.1.14 (#2827)

**Comment by hoodmane** [comment]
Maybe we should backport the patch from emscripten 3.15.

**Comment by rth** [comment]
BTW, `du` measure the size in the filesystem, not the file size and would not be accurate in a number of cases (https://askubuntu.com/a/1163848), so we should probably stop using it for this metric.

**Comment by rth** [comment]
Overall +1 to unvendor it, if there are no major objections from Christian. Particularly now that you added more explicit exception errors for unvendored / missing stlib packages in https://github.com/pyodide/pyodide/pull/2973

There are probably not that many packages that need sqlite3 compared to the total number, and for the rest an extra 1.5MB (probably 0.5MB compressed) is a lot both in load and WASM compilation time.

**Comment by tiran** [comment]
I found a problem with Emscripten ports, see https://github.com/emscripten-core/emscripten/issues/17675

To work around the problem, do ``embuilder build --pic sqlite3`` and add this to Modules/Setup.local

```
*shared*
_sqlite3 _sqlite/blob.c _sqlite/connection.c _sqlite/cursor.c _sqlite/microprotocols.c _sqlite/module.c _sqlite/prepare_protocol.c _sqlite/row.c _sqlite/statement.c _sqlite/util.c -lsqlite3
```

**Comment by ryanking13** [comment]
> I found a problem with Emscripten ports, see [emscripten-core/emscripten#17675](https://github.com/emscripten-core/emscripten/issues/17675)

I was always wondering why `-sUSE_...` doesn't really works when linking packages, and that was the reason... thanks for the info Christian!

File: docs/project/changelog.md [hunk 1]
```diff
@@ -24,6 +24,11 @@ substitutions:
   `append`, and `pop` for `JsProxy` of Javascript arrays.
   {pr}`2970`
 
+- {{ Breaking }} \_sqlite3 stdlib module is not unvendored by default.
+  Now it needs to be maually loaded by {any}`pyodide.loadPackage`
+  or {any}`micropip.install`.
```
Comment (hoodmane) [inline docs/project/changelog.md:29]:
```suggestion
- {{ Breaking }} Unvendored the \_sqlite3 module from the standard library.
  Before `sqlite3` was included by default. Now it needs to be loaded with 
  {any}`pyodide.loadPackage` or {any}`micropip.install`.
```

**Comment by hoodmane** [comment]
Do we also want to unvendor the `sqlite` python package? It is 143kb uncompressed.

**Comment by ryanking13** [comment]
> Do we also want to unvendor the `sqlite` python package? It is 143kb uncompressed.

Right, probably we should do that for `ssl` and `lzma` too.

**Review by hoodmane — APPROVED** [review]
This looks very nice. Thanks @ryanking13!

**Comment by ryanking13** [comment]
Thanks for the reviews!

**Pull request merged**